### DISCUSSION TITLE
ch3/ch3-06

### DISCUSSION THREAD

**Discussion by giscus** [gopl-zh/gopl-zh.github.com#220] [General]
# ch3/ch3-06

<The Go Programming Language>中文版

https://gopl-zh.github.io/ch3/ch3-06.html

**Comment by newarabbit** [comment]
3.13
新手提问：为什么我跑的时候到ZIB和YIB才溢出，不应该到TIB就溢出吗 32位
```
const (    
    KB = 1 << (10 * iota)    
    MB = KB << 10    
    GB = MB << 10    
    TB = GB << 10    
    PB = TB << 10    
    EB = PB << 10    
    ZB = EB << 10    
    YB = ZB << 10 
)
```

**Reply by diantaowang** [reply]
64位系统下，go int类型为64-bit

**Reply by newarabbit** [reply]
明白了 谢谢

**Comment by Apahno** [comment]
i was here

**Comment by hongjiala** [comment]
3.13package main

import (
	"fmt"
	"math/big"
)

func main() {
	// 定义大整数
	KB := big.NewInt(1 << 10)
	MB := new(big.Int).Lsh(KB, 10)
	GB := new(big.Int).Lsh(MB, 10)
	TB := new(big.Int).Lsh(GB, 10)
	PB := new(big.Int).Lsh(TB, 10)
	EB := new(big.Int).Lsh(PB, 10)
	ZB := new(big.Int).Lsh(EB, 10)
	YB := new(big.Int).Lsh(ZB, 10)

	// 打印大整数
	fmt.Println("KB:", KB)
	fmt.Println("MB:", MB)
	fmt.Println("GB:", GB)
	fmt.Println("TB:", TB)
	fmt.Println("PB:", PB)
	fmt.Println("EB:", EB)
	fmt.Println("ZB:", ZB)
	fmt.Println("YB:", YB)
}
